wasm/validator/src/lib.rs

```diff
@@ -112,8 +112,15 @@ macro_rules! declare_tokens {
 
 /// Validator of Iroha operations
 pub trait Validate: Visit {
-    /// Get all [`PermissionTokenDefinition`]'s defined by validator.
-    fn permission_tokens() -> Vec<PermissionTokenDefinition>;
+    /// Migrate previous validator to the current version.
+    ///
+    /// This function should be called by `migrate` entrypoint,
+    /// which will be called by Iroha only once just before upgrading validator.
+    ///
+    /// # Errors
+    ///
+    /// Concrete errors are specific to the implementation.
+    fn migrate() -> MigrationResult;
```

Comment:
I get the feeling `migrate` doesn't belong into `Validate` trait. Let's have a short discussion on our naming. We should adjust our naming a little bit since we've moved from validator to executor. Can you give a thought about the naming conventions

Reply:
I agree that `migrate` doesn't look nice in `Validate` trait. My hope is that we eventually rename it to `Execute`

Reply:
As well as all *validator* related terminology will be renamed to *executor* related.